train.py

```diff
@@ -98,50 +101,93 @@ def train(params):
     # Load hgn from parameters to deice
     hgn = load_hgn(params=params, device=device, dtype=dtype)
 
-    # Dataloader
-    dataset_len = params["optimization"]["epochs"] * params["optimization"][
-        "batch_size"]
-    seed = None if params["dataset"]["random"] else 0
-    trainDS = EnvironmentSampler(
-        environment=env,
-        dataset_len=dataset_len,
-        number_of_frames=params["rollout"]["seq_length"],
-        delta_time=params["rollout"]["delta_time"],
-        number_of_rollouts=params["optimization"]["batch_size"],
-        img_size=params["dataset"]["img_size"],
-        color=params["rollout"]["n_channels"] == 3,
-        noise_std=params["dataset"]["noise_std"],
-        radius_bound=params["dataset"]["radius_bound"],
-        world_size=params["dataset"]["world_size"],
-        seed=seed,
-        dtype=dtype)
-    # Dataloader instance test, batch_mode disabled
-    data_loader = torch.utils.data.DataLoader(trainDS,
-                                              shuffle=False,
-                                              batch_size=None)
+    # Either generate data on-the-fly or load the data from disk
+    data_loader = None
+    if params["dataset"]["on_the_fly_data"]:
+        dataset_len = params["optimization"]["epochs"] * params[
+            "optimization"]["batch_size"]
```

Comment:
Should we just have `dataset_len` as parameter in the yaml file? It should be independent of the batch size we use. It also makes it difficult to understand how many samples will be used. For example, in `default.yaml`, `epochs=5` and `batch_size=16`, so `dataset_len=80` rollouts. But if `batch_size=32` then the training steps be doubled.